lib/NeuraDialect/Mapping/mapping_util.cpp

```diff
@@ -416,143 +417,158 @@ bool mlir::neura::tryRouteDataMove(Operation *mov_op, MappingLoc src_loc,
                                    MappingLoc dst_loc, bool is_backward_move,
                                    const MappingState &state,
                                    std::vector<MappingLoc> &path_out) {
-  // Specially handles the case where src and dst are the same tile.
-  if (src_loc.resource == dst_loc.resource) {
-    // When the source and destination are on the same tile, we still need to
-    // allocate registers for data movement.
-    Tile *tile = dyn_cast<Tile>(src_loc.resource);
-    if (!tile) {
-      llvm::errs() << "[tryRouteDataMove] Source is not a tile\n";
-      return false;
-    }
+  assert(path_out.empty() && "Output path should be empty");
+
+  // Gets the source tile and destination tile.
+  Tile *src_tile = dyn_cast<Tile>(src_loc.resource);
+  Tile *dst_tile = dyn_cast<Tile>(dst_loc.resource);
+
+  assert(src_tile && dst_tile &&
+         "Source and destination locations must be tiles");
 
-    // Defines the time window for the register allocation.
+  // Calculates the deadline time step (adds II for backward moves).
+  int exclusive_deadline_step = dst_loc.time_step;
+  if (is_backward_move) {
+    exclusive_deadline_step += state.getII();
+  }
+
+  llvm::outs() << "[tryRouteDataMove] Routing from Tile#" << src_tile->getId()
+               << " @t=" << src_loc.time_step << " to Tile#"
+               << dst_tile->getId() << " @t=" << exclusive_deadline_step
+               << "\n";
+
+  // Special case: source tile and destination tile are the same.
+  if (src_tile == dst_tile) {
+    // Uses register as routing resource within the same tile.
     int start_time = src_loc.time_step;
-    int exclusive_end_time = dst_loc.time_step;
-
-    // For backward moves, we need to adjust the end time.
-    if (is_backward_move) {
-      exclusive_end_time += state.getII();
-    }
-
-    // Finds a register that is available during start_time to
-    // exclusive_end_time.
-    Register *reg =
-        getAvailableRegister(state, tile, start_time, exclusive_end_time);
-    if (!reg) {
-      llvm::errs() << "[tryRouteDataMove] No available register found for "
-                   << "tile: " << tile->getId()
-                   << " from time step: " << start_time
-                   << " to time step: " << exclusive_end_time << "\n";
+    int exclusive_end_time = exclusive_deadline_step;
```

Comment:
Why do we need `exclusive_end_time`, cannot just use `exclusive_deadline_step` here?